llvm/include/llvm/IR/IntrinsicsSPIRV.td

```diff
@@ -86,6 +86,7 @@ let TargetPrefix = "spv" in {
   def int_spv_dot4add_i8packed : DefaultAttrsIntrinsic<[llvm_i32_ty], [llvm_i32_ty, llvm_i32_ty, llvm_i32_ty], [IntrNoMem]>;
   def int_spv_dot4add_u8packed : DefaultAttrsIntrinsic<[llvm_i32_ty], [llvm_i32_ty, llvm_i32_ty, llvm_i32_ty], [IntrNoMem]>;
   def int_spv_wave_active_countbits : DefaultAttrsIntrinsic<[llvm_i32_ty], [llvm_i1_ty], [IntrConvergent, IntrNoMem]>;
+  def int_spv_wave_any : DefaultAttrsIntrinsic<[llvm_i1_ty], [llvm_i1_ty], [IntrConvergent, IntrNoMem]>;
```

Comment:
Shall IntrNoMem be removed?
(readlane, active_countbits have it, but maybe that's wrong).

AFAIK is IntrNoMem is added, compiler can assume there are no other side effect. But since this depends on other lanes, isn't that equivalent to IntrReadMem?

Reply:
@inbelic says its fine to keep here https://github.com/llvm/llvm-project/pull/115902#discussion_r1849054561

Reply:
Happy to be corrected. My understanding was that `IntrConvergent` was used to model the cross-lane side-effect, and we can depend on that to ensure no optimizations would cause a bad reordering/folding. So we can then use `IntrNoMem` to model that there are no other (memory) side-effects.

cc @farzonl @bogner

Reply:
My understanding was `IntrConvergent` prevented optimizations to move this instruction into another control-flow, but could change the order in a given BB/flow, but from the few tests I did, it seems those are not optimized anyway (even dead-stores).
So for now let's go with `IntrConvergent` and `IntrNoMem`, and if we have a repro/issue in the future, we'll know for sure.